### ISSUE TITLE
SPARQL test suite compliance

### ISSUE THREAD

**Issue by rubensworks** [comunica/comunica#287]
This issue lists all current failures from the SPARQL 1.1 test suite,
categorized by type.

## SPARQL Syntax

* [ ] syntax-oneof-03.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_33)
* [ ] syntax-bindings-02a.rq with VALUES clause (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_35a)
* [ ] syntax-bindings-03a.rq with VALUES clause (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_36a)
* [ ] syntax-construct-where-02.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_42)
* [ ] syn-bad-01.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_43)
* [ ] syn-bad-02.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_44)
* [ ] syn-bad-03.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_45)
* [ ] syntax-BINDscope6.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_60)
* [ ] syntax-BINDscope7.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_61a)
* [ ] syntax-BINDscope8.rq (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_62a)
* [ ] syntax-SELECTscope2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/syntax-query/manifest#test_65)

## Sub-queries

* [ ] Expression has variable that may be unbound (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp07)
* [ ] sq11 - Subquery limit per resource (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery11)
* [ ] sq12 - Subquery in CONSTRUCT with built-ins (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery12)
* [ ] sq13 - Subqueries don't inject bindings (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery13)
* [ ] sq14 - limit by resource (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery14)

## Values

* [ ] Post-query VALUES with subj-var, 1 row (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values1)
* [ ] Post-query VALUES with 2 obj-vars, 1 row (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values3)
* [ ] Post-query VALUES with 2 obj-vars, 1 row with UNDEF (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values4)
* [ ] Post-query VALUES with 2 obj-vars, 2 rows with UNDEF (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values5)
* [ ] Post-query VALUES with (OPTIONAL) obj-var, 1 row (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values7)
* [ ] Post-query VALUES with subj/obj-vars, 2 rows with UNDEF (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bindings/manifest#values8)

## Construct

* [ ] constructwhere02 - CONSTRUCT WHERE (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/construct/manifest#constructwhere02)
* [ ] constructwhere03 - CONSTRUCT WHERE (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/construct/manifest#constructwhere03)
* [ ] constructwhere04 - CONSTRUCT WHERE (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/construct/manifest#constructwhere04)

## Property paths

* [ ] (pp02) Star path (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp02)
* [ ] (pp12) Variable length path and two paths to same target node (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp12)
* [ ] (pp14) Star path over foaf:knows (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp14)
* [ ] (pp16) Duplicate paths and cycles through foaf:knows* (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp16)
* [ ] (pp28a) Diamond, with loop -- (:p/:p)? (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp28a)
* [ ] (pp30) Operator precedence 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp30)
* [ ] (pp32) Operator precedence 3 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp32)
* [ ] (pp33) Operator precedence 4 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp33)
* [ ] (pp34) Named Graph 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp34)
* [ ] (pp35) Named Graph 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/property-path/manifest#pp35)

## Expressions

__Blocked by #39.__

* [ ] COUNT 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg01)
* [ ] COUNT 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg02)
* [ ] COUNT 3 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg03)
* [ ] COUNT 4 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg04)
* [ ] COUNT 5 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg05)
* [ ] COUNT 6 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg06)
* [ ] COUNT 7 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg07)
* [ ] COUNT 8 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg08)
* [ ] COUNT 8b (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg08b)
* [ ] COUNT 9 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg09)
* [ ] COUNT 10 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg10)
* [ ] COUNT 11 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg11)
* [ ] COUNT 12 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg12)
* [ ] GROUP_CONCAT 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-groupconcat-01)
* [ ] GROUP_CONCAT 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-groupconcat-02)
* [ ] GROUP_CONCAT with SEPARATOR (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-groupconcat-03)
* [ ] SUM (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-sum-01)
* [ ] SUM with GROUP BY (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-sum-02)
* [ ] AVG (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-avg-01)
* [ ] AVG with GROUP BY (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-avg-02)
* [ ] MIN (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-min-01)
* [ ] MIN with GROUP BY (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-min-02)
* [ ] MAX (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-max-01)
* [ ] MAX with GROUP BY (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-max-02)
* [ ] SAMPLE (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-sample-01)
* [ ] Error in AVG (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-err-01)
* [ ] Protect from error in AVG (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-err-02)
* [ ] agg empty group (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/aggregates/manifest#agg-empty-group2)
* [ ] bind01 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind01)
* [ ] bind02 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind02)
* [ ] bind03 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind03)
* [ ] bind04 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind04)
* [ ] bind05 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind05)
* [ ] bind06 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind06)
* [ ] bind07 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind07)
* [ ] bind08 - BIND (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind08)
* [ ] bind10 - BIND scoping - Variable in filter not in scope (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind10)
* [ ] bind11 - BIND scoping - Variable in filter in scope (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/bind/manifest#bind11)
* [ ] Exists with one constant (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists01)
* [ ] Exists with ground triple (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists02)
* [ ] Exists within graph pattern (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists03)
* [ ] Nested positive exists (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists04)
* [ ] Nested negative exists in positive exists (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists05)
* [ ] STRDT() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strdt01)
* [ ] STRDT(STR()) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strdt02)
* [ ] STRDT() TypeErrors (updated for RDF 1.1) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strdt03-rdf11)
* [ ] STRLANG() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strlang01)
* [ ] STRLANG(STR()) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strlang02)
* [ ] STRLANG() TypeErrors (updated for RDF 1.1) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strlang03-rdf11)
* [ ] isNumeric() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#isnumeric01)
* [ ] ABS() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#abs01)
* [ ] CEIL() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#ceil01)
* [ ] FLOOR() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#floor01)
* [ ] ROUND() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#round01)
* [ ] CONCAT() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#concat01)
* [ ] CONCAT() 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#concat02)
* [ ] SUBSTR() (3-argument) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#substring01)
* [ ] SUBSTR() (2-argument) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#substring02)
* [ ] STRLEN() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#length01)
* [ ] UCASE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#ucase01)
* [ ] LCASE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#lcase01)
* [ ] ENCODE_FOR_URI() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#encode01)
* [ ] STRSTARTS() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#starts01)
* [ ] STRENDS() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#ends01)
* [ ] plus-1-corrected (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#plus-1-corrected)
* [ ] plus-2-corrected (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#plus-2-corrected)
* [ ] MD5() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#md5-01)
* [ ] MD5() over Unicode data (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#md5-02)
* [ ] SHA1() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha1-01)
* [ ] SHA1() on Unicode data (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha1-02)
* [ ] SHA256() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha256-01)
* [ ] SHA256() on Unicode data (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha256-02)
* [ ] SHA512() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha512-01)
* [ ] SHA512() on Unicode data (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#sha512-02)
* [ ] MINUTES() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#minutes)
* [ ] SECONDS() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#seconds)
* [ ] HOURS() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#hours)
* [ ] MONTH() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#month)
* [ ] YEAR() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#year)
* [ ] DAY() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#day)
* [ ] TIMEZONE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#timezone)
* [ ] TZ() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#tz)
* [ ] BNODE(str) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#bnode01)
* [ ] BNODE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#bnode02)
* [ ] IN 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#in01)
* [ ] IN 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#in02)
* [ ] NOT IN 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#notin01)
* [ ] NOT IN 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#notin02)
* [ ] NOW() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#now01)
* [ ] RAND() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#rand01)
* [ ] IRI()/URI() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#iri01)
* [ ] IF() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#if01)
* [ ] IF() error propogation (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#if02)
* [ ] COALESCE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#coalesce01)
* [ ] STRBEFORE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strbefore01a)
* [ ] STRBEFORE() datatyping (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strbefore02)
* [ ] STRAFTER() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strafter01a)
* [ ] STRAFTER() datatyping (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#strafter02)
* [ ] REPLACE() (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#replace01)
* [ ] REPLACE() with overlapping pattern (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#replace02)
* [ ] REPLACE() with captured substring (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#replace03)
* [ ] UUID() pattern match (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#uuid01)
* [ ] STRUUID() pattern match (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/functions/manifest#struuid01)
* [ ] Group-1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group01)
* [ ] Group-3 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group03)
* [ ] Group-4 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group04)
* [ ] Group-5 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group05)
* [ ] Group-6 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group06)
* [ ] Group-7 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/grouping/manifest#group07)
* [ ] Subsets by exclusion (NOT EXISTS) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#subset-by-exclusion-nex-1)
* [ ] Medical, temporal proximity by exclusion (NOT EXISTS) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#temporal-proximity-by-exclusion-nex-1)
* [ ] Calculate which sets are subsets of others (include A subsetOf A) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#subset-01)
* [ ] Calculate which sets are subsets of others (exclude A subsetOf A) (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#subset-02)
* [ ] Calculate which sets have the same elements (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#set-equals-1)
* [ ] Calculate proper subset (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#subset-03)
* [ ] Positive EXISTS 1 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#exists-01)
* [ ] Positive EXISTS 2 (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#exists-02)
* [ ] Subtraction with MINUS from a fully bound minuend (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#full-minuend)
* [ ] Subtraction with MINUS from a partially bound minuend (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/negation/manifest#partial-minuend)
* [ ] Expression is equality (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp01)
* [ ] Expression raise an error (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp02)
* [ ] Reuse a project expression variable in select (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp03)
* [ ] Reuse a project expression variable in order by (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp04)
* [ ] Expression may return no value (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp05)
* [ ] Expression has undefined variable (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/project-expression/manifest#projexp06)
* [ ] sq02 - Subquery within graph pattern, graph variable is bound (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery02)
* [ ] sq08 - Subquery with aggregate (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery08)
* [ ] sq10 - Subquery with exists (http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery10)

**Comment by rubensworks** [comment]
Once this is done, we should publish our EARL report on https://github.com/w3c/rdf-tests/

**Comment by wschella** [comment]
'Exists within graph pattern' (<http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists03>) fails with following output.

```shell
✖ Exists within graph pattern
  Checks that exists is interpreted within named graph
  Error: Cannot resolve relative IRI exists02.ttl because no base IRI was set.
  More info: http://www.w3.org/2009/sparql/docs/tests/data-sparql11/exists/manifest#exists03
```

This is the query that is passed.

```SPARQL
prefix ex: <http://www.example.org/>

select * where {
graph <exists02.ttl> { 
  ?s ?p ex:o1
  filter exists { ?s ?p ex:o2 } 
}

}
```

**Comment by rubensworks** [comment]
exists03 _may_ assume that `BASE <...>` is prefixed into the SPARQL query, but I'll need to look into that a bit further...

**Comment by rubensworks** [comment]
Current status:

> 214 / 268 tests succeeded!

**Comment by rubensworks** [comment]
@wschella I fixed the baseIRI issue (commit incoming).
The test does not pass yet though, the following error is shown now:

 ```
  Expected: [QueryResultBindings:
    Variables: [
  "?p",
  "?s"
]
    Bindings:  [
  {
    "?s": {
      "value": "http://www.example.org/b"
    },
    "?p": {
      "value": "http://www.example.org/p"
    }
  }
]
]

  Got:
 [QueryResultBindings:
    Variables: [
  "?p",
  "?s"
]
    Bindings:  []
]
```

I assume this is an error on your end.

**Comment by wschella** [comment]
Great, thanks! This is something i can work with.

**Comment by simonvbrae** [comment]
It seems that the constructorArguments for AbstractPath don't have an id which might be contributing to some of the problems with the property paths tests.
